### PR TITLE
Move definition of widgethelper::mapPopupToScreen() into .cpp file

### PR DESCRIPTION
**Pull request by uklotzde**
...to fix missing #include directives and to reduce dependencies.

- More concise implementation using math_min()
- Use math_max(0, ...) to prevent that the upper left corner is displayed outside the screen area if the size of the popup menu exceeds the size of the screen

This PR fixes a technical issue that was detected in #2793.

### CODE REVIEW

**Comment by uklotzde** [comment]
Let's see if the rebase fixes the pre-commit-hook warnings.

**Comment by uklotzde** [comment]
> Let's see if the rebase fixes the pre-commit-hook warnings.

Nope. But I won't open a new PR to get rid of them. Thus the CI "failure" is unrelated.

File: src/util/widgethelper.cpp [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+namespace widgethelper {
+
+QPoint mapPopupToScreen(
+        const QWidget& widget,
+        const QPoint& popupUpperLeft,
+        const QSize& popupSize) {
+    const auto screenSize = widget.windowHandle()->screen()->size();
+    const auto adjustedX = math_max(0,
+            math_min(
```
Comment (Be-ing) [inline src/util/widgethelper.cpp:18]: We have `math_clamp` in src/util/math.h for this.

File: src/widget/woverview.cpp [hunk 2]
```diff
@@ -531,7 +531,7 @@ void WOverview::mousePressEvent(QMouseEvent* e) {
             if (pHoveredCue != nullptr) {
                 m_pCueMenuPopup->setTrackAndCue(m_pCurrentTrack, pHoveredCue);
                 QPoint cueMenuTopLeft = mixxx::widgethelper::mapPopupToScreen(
-                        windowHandle()->screen()->size(),
+                        *this,
```
Comment (Be-ing) [inline src/widget/woverview.cpp:534]: Good idea to move this into the utility function.

**Comment by Be-ing** [comment]
> Use math_max(0, ...) to prevent that the upper left corner is displayed outside the screen area if the size of the popup menu exceeds the size of the screen

Is this check really needed? If the popup is too big for the screen, why should this utility function assume that showing the top left of the popup is the best fallback? Anyway, I don't know in what situation this might actually happen.

**Comment by uklotzde** [comment]
> > Use math_max(0, ...) to prevent that the upper left corner is displayed outside the screen area if the size of the popup menu exceeds the size of the screen
> 
> Is this check really needed? If the popup is too big for the screen, why should this utility function assume that showing the top left of the popup is the best fallback? Anyway, I don't know in what situation this might actually happen.

It is a viable assumption for left-to-right/top-to-bottom writing style languages.

**Comment by uklotzde** [comment]
I have reverted the last commit and added a comment why we can't use math_clamp() here.

**Pull request merged**